### ISSUE TITLE
How to confirms publish

### ISSUE THREAD

**Issue by matboulos** [progresspoint/txamqp#1]
Hi,
I could not find a way to handle publish confirms returned by the broker after a basic_publish operation with madatory field set to True.
I've tried to yield the deferred returned by channel.basic_publish(..., mandatory=True) without any success (return None).
I work on RabbitMq 3.5.3-1 
same result loading amqp0-8.stripped.rabbitmq.xml or amqp0-9-1.stripped.xml spec file.
Do i have to set something on the channel to enable confirms.
Could you make a sample available please.

Many thanks

Mat

**Comment by ei-grad** [comment]
You have to enable publish confirms on the channel using channel.confirm_select() method, and then all basic_publish() calls on this channel would return deferreds instead of None. Deferred would be resolved with True if message was published normally, or False if it can't.

**Comment by ei-grad** [comment]
Btw, mandatory flag is not about publish confirms feature. It is the standard AMQP feature which just returns the whole message back to producer with Return method in case if there are no queues bound to the exchange and routing key specified in basic_publish, and it should be handled separately from publish confirms.

**Comment by matboulos** [comment]
You're right Andrew, and i need the Basic.return info too, i will try the confirm_select on the channel.
Thanks for your help !

**Comment by matboulos** [comment]
Hello Andrew,
I've made some tests today in channel's "confirm mode" using channel.confirm_select(), but i am wondering if Basic.ack's multiple flag is managed by the library.

I've put a sample script on gists to illustrate this.
https://gist.github.com/matboulos/e2aaaa61c38e90148c0d#file-txamqp-ack-test-py.

With short msg_rate (eg : python txamqp-ack-test.py 0.01), some deferreds are never called back.
I think it's because RabbitMq acknowledge multiple messages in one Basic.ack message (https://www.rabbitmq.com/confirms.html).
If so, how can i consider that a message is acked by the broker if the underlying deferred never fire his callback ?

CMIIW

**Comment by ei-grad** [comment]
Wow. Good catch, thanks. This is a bug, I just didn't notice the info about multiple flag in confirms docs. Will see how to fix it...
